documentation/API_documentation/QoD_API.md

```diff
@@ -0,0 +1,178 @@
+<br>
+
+#### QoD Delete QoS Session
+
+| **Deleting QoS session** |
+| ---------------------------- |
+| **HTTP Request**<br>  DELETE\<base-url>/qod-api/v0/sessions/{sessionId}<br>**Query Parameters**<br>  No query parameters are defined.<br>**Path Parameters**<br>  sessionId: Session ID that needs to be terminated.<br>**Request Body Parameters**<br>  No request body parameters are defined.<br><br>**Response**<br> **204:** Session deleted<br> **401:** Un-authorized. <br> **403:** Forbidden.<br> **404:** Session not found.<br> **503:** Service temporarily unavailable.|
+
+### 4.4 Errors
+
+Since CAMARA QoD API is based on REST design principles and blueprints, well defined HTTP status codes and families specified by community are followed [[4]](#4).
+
+Details of HTTP based error/exception codes for the QoD API are described in Section 4.3 of each API REST based method.
+Following table provides an overview of common error names, codes, and messages applicable to QoD API.
+
+| No | Error Name | Error Code | Error Message |
+| --- | ---------- | ---------- | ------------- |
+|1	|400 |	INVALID_INPUT |	"Expected property is missing: ueId.msisdn" |
+|2	|400 |	INVALID_INPUT |	"Expected property is missing: ueId.ipv4addr" |
+|3	|400 |	INVALID_INPUT |	"Expected property is missing: ueId.ipv4addr or ueId.ipv6addr" |
+|4	|400 |	INVALID_INPUT |	"Expected property is missing: uePorts" |
+|5	|400 |	INVALID_INPUT |	"Expected property is missing: qos" |
+|6	|400 |	INVALID_INPUT |	"Ranges not allowed: uePorts" |
+|7	|401 |	UNAUTHORIZED |	"No authorization to invoke operation" |
+|8	|403 |	FORBIDDEN |	"Operation not allowed" |
+|9	|404 |	NOT_FOUND |	"Session Id does not exist" |
+|10	|409 |	CONFLICT |	"Another session is created for the same UE" |
+|11	|500 |	INTERNAL |	"Session could not be created" |
+|12	|503 |	SERVICE_UNAVAILABLE |	"Service unavailable" |
+
+### 4.5 Policies
+
+N/A
+
+### 4.6 Code Snippets
+<br>
+<span class="colour" style="color:rgb(36, 41, 47)">Snippet 1, elaborates REST based API call with "*curl"* to create a QoS session for sample streaming service with following parameters: </span>
+
+* QoS session with QoS-profile "QOS_E" mapping,
+* App-Flow is specified for UE-Terminal identifier (ueId=2001:db8:85a3:8d3:1319:8a2e:370:7344), Application server identifier (asId=54.204.25.0/28) and Port number (asPorts=33001).
```

Comment:
Looks complicated to have an IPv6 address as UE identifier and an IPv4 address as an Application Server Id. This may raise a bunch of technical questions.

Reply:
I suggest that it is an actual, even typical case: Let's assume that the mobile network is internally using only IPv6 (or only the IPv6 address is unique to identify the UE), and the application server is offering the service on IPv4 address. The mobile network is then using CGNAT. I agree that there are a several technical questions which need to be addressed (partly candidates for the FAQ section, partly open issue, e.g. #34).

Reply:
You mean, the CGNAT is doing NAT64? I suggest to add some more description around the network assumptions. The Application Developers need to understand the concepts of the CAMARA QoD funtion.

Reply:
@tlohmar: as said, this is a perfect example for an FAQ. The challenge is that there is no general explanation possible as operator network implementation vary. Do you see how this can be done in a general FAQ entry here in CAMARA?